### ISSUE TITLE
Validation and test sets in S3DIS evaluation

### ISSUE THREAD

**Issue by chaitjo** [HuguesTHOMAS/KPConv-PyTorch#122]
https://github.com/HuguesTHOMAS/KPConv-PyTorch/blob/cf1f06381ef062344b68286e3d5034b1aa94aecd/test_models.py#L170

I had a quick Q. -- why are S3DIS models always evaluated on the validation set?

Or is validation == test in this case? Is it the Area-5 split?

**Comment by chaitjo** [comment]
Another quick Q. -- for reporting the final result for a run on S3DIS, the correct number to report will be the latest mIoU printed after 'Confusion on full clouds'. Am I correct?

**Comment by HuguesTHOMAS** [comment]
> why are S3DIS models always evaluated on the validation set?

S3DIS does not have a private test set like the other benchmarks (Scannet, Semantic3D, NPM3D). In their original paper, they selected Area5 as the test set and the rest as the training set. In that case, "test set" and "validation set" are the same as the labels for Area5 are available to anyone.
Note that some papers use k-fold validation instead of having only Area5 as test/validation set. Both are good, but I chose to follow the original paper guidelines.

>  for reporting the final result for a run on S3DIS, the correct number to report will be the latest mIoU printed after 'Confusion on full clouds'. Am I correct?

Yes, you have to report the scores from 'Confusion on full clouds'. If I remember correctly the first score is the mean IoU and then you have all the class IoU. You can verify that by averaging all the class IoU it should be the same number as the mean IoU.
